### PR TITLE
annotate: don't re-read mbentry every time if mailbox is open

### PR DESCRIPTION
**Pull request by ksmurchison**
I think this is a simpler/cleaner solution than #4981

### CODE REVIEW

File: imap/annotate.c [hunk 1]
```diff
@@ -1229,7 +1215,7 @@ static int annotatemore_findall_full(const char *pattern, /* internal */
         struct findall_data data = { .mbentry = mbentry, .is_exactmatch = 1 };
         r = _findall(&data, &frock);
     }
-    else if (!*pattern) {
+    else if (!pattern) {
```
Comment (brong) [inline imap/annotate.c:1218]: I'm not sure if we should be looking for BOTH here - NULL or blank string.

File: imap/annotate.c [hunk 2]
```diff
@@ -1204,19 +1202,7 @@ static int annotatemore_findall_full(const char *pattern, /* internal */
...
-        mboxname = NULL;
-        pattern = "";
-    }
-
-    if (mailbox || mboxname) {
-        if (mailbox) mboxname = mailbox_name(mailbox);
-        r = mboxlist_lookup_allow_all(mboxname, &mbentry, NULL);
-        if (r) goto out;
-    }
+    if (mailbox) mbentry = mailbox_mbentry(mailbox);
```
Comment (brong) [inline imap/annotate.c:1205]: Should we lift this into all the callers as well?

**Review by brong — COMMENTED** [review]
A couple comments - I'm not sure if this goes quite far enough to normalising things, but looks generally good :)

File: imap/annotate.c (hunk 1 above)
Comment (ksmurchison) [reply imap/annotate.c:1218]: Empty string in the previous code was used to signal server entries, but that signaling has changed.  Having said that I have to issue with checking for both.

File: imap/annotate.c (hunk 2 above)
Comment (ksmurchison) [reply imap/annotate.c:1205]: I think I had in annotatemore _findall_mailbox() and moved it out.  I don't recall why and it probably wasn't a good reason.  It probably should be there.

**Review by rsto — APPROVED** [review]

**Pull request merged**